azkaban-exec-server/src/main/java/azkaban/dag/Node.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright 2017 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package azkaban.dag;
+
+import static java.util.Objects.requireNonNull;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Node in a DAG: Directed acyclic graph.
+ */
+class Node {
+
+  private final String name;
+
+  private final NodeProcessor nodeProcessor;
+
+  // The nodes that this node depends on.
+  private final List<Node> parents = new ArrayList<>();
```

Comment:
This terminology can be confusing because if this is a sub-flow, parent should be the enclosing flow. So this field could be renamed to "dependencies", and below "children" -> "dependents"?

Reply:
As mentioned earlier, at this layer, there is no concept of sub-flows.